### ISSUE TITLE
Running CRC behind proxy

### ISSUE THREAD

**Issue by piggyvenus** [code-ready/crc#337]
Export `HTTP_PROXY` and `HTTPS_PROXY`.
Getting error:
```
sudo crc setup
INFO Caching oc binary                            
ERRO Error occurred: Download failed: Head https://mirror.openshift.com/pub/openshift-v4/clients/oc/latest/macosx/oc.tar.gz: dial tcp: lookup mirror.openshift.com on x.x.x.x:53: no such host 
FATA Not able to download oc Download failed: Head https://mirror.openshift.com
/pub/openshift-v4/clients/oc/latest/macosx/oc.tar.gz: dial tcp: lookup mirror.openshift.com on x.x.x.x:53: no such host 
```

**Comment by gbraad** [comment]
Currently not possible. We will add proxy support when OpenShift has added support on cluster level.

**Comment by cfergeau** [comment]
For what it's worth, the cavaliercoder/grab module that we use for our downloads seems to be able to use proxies, so I would have expected oc/machine driver downloads to work. Needs more testing and debugging ;)

**Comment by praveenbommali** [comment]
We're not able to access in  corporate environment and  getting the below issue  
ERRO Download failed: Head https://mirror.openshift.com/pub/openshift-v4/clients/oc/latest/windows/oc.zip: dial tcp 54.173.18.88:443: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
FATA Unable to download oc Download failed: Head https://mirror.openshift.com/pub/openshift-v4/clients/oc/latest/windows/oc.zip: dial tcp 54.173.18.88:443: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.

**Comment by praveenbommali** [comment]
How to  set the proxy while  running 
`>> crc setup  `

**Comment by praveenkumar** [comment]
@praveenbommali we still don't have proxy support for crc right now, it is on our high priority list and will be start working soon on that.

**Comment by cfergeau** [comment]
https://github.com/code-ready/crc/issues/509 would workaround these issues with downloading oc/machine-drivers  when a proxy is involved

**Comment by cfergeau** [comment]
Another issue we will have when adding proxy support is `hosts -R 3 quay.io` which we currently use, but will need to change to something different.

**Comment by jepries** [comment]
Sorry to rehash this, it does look like in your 1.2.0 bundle, some of this has been fixed.  I was able to run 

    crc setup

with proxies just fine (see https://github.com/code-ready/crc/issues/864)

Once I set "export NO_PROXY=.testing" I can run 

    crc start

no problem.

Once I set NO_PROXY to .testing inside my Firefox Connections Settings, I can get to the OpenShift console perfectly after running

    crc console

Here is my current issue. 

    Failed to pull image "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:a16c3892ffd8ebc28677fec63db9311adfeb99693b99e10fef16573e861f6bbb": rpc error: code = Unknown desc = pinging docker registry returned: Get https://quay.io/v2/: dial tcp 3.230.48.144:443: i/o timeout

 I'm trying to run httpd-example project, and the build keeps getting stuck.  I've added HTTP_PROXY, HTTPS_PROXY env settings to dc (i know know that comes later, i'm a beginner), then found bc, added them there, canceled/restarted the builds, and no joy.

I haven't yet found the secret sauce yet, but I want to stop searching if this is related to this issue you guys are working on.
